### ISSUE TITLE
c_sdk API Inferences error

### ISSUE THREAD

**Issue by DDGRCF** [open-mmlab/mmdeploy#750]
### Envs
Ubuntu22.04 G++-9 GCC-9

### Convert  Command
```shell
python tools/deploy.py configs/mmdet/detection/detection_onnxruntime_dynamic.py"  $my_config_path $my_config_ckpt ... --device cpu
```
### Inference Error Command 
```shell 
./object_detection cpu  $model_path $image_path 
```
### Error
```shell
......
[2022-07-14 20:29:34.812] [mmdeploy] [info] [common.h:29] config: {
  "component": "ResizeBBox",
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "input": [
    "prep_output",
    "infer_output"
  ],
  "module": "mmdet",
  "name": "postprocess",
  "output": [
    "post_output"
  ],
  "params": {
    "nms": {
      "iou_threshold": 0.65,
      "type": "nms"
    },
    "score_thr": 0.01
  },
  "type": "Task"
}

[2022-07-14 20:29:34.812] [mmdeploy] [warning] [bulk.h:39] fallback Bulk implementation
[2022-07-14 20:29:34.813] [mmdeploy] [warning] [bulk.h:39] fallback Bulk implementation
/opt/rh/devtoolset-9/root/usr/include/c++/9/bits/stl_vector.h:1042: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = int; _Alloc = std::allocator<int>; std::vector<_Tp, _Alloc>::reference = int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
[1]    124742 IOT instruction (core dumped)  ./object_detection cpu ${model_path} ${img_path}

```

**Comment by lzhangzz** [comment]
Sorry, can't reproduce the problem with
```
​./object_detection cpu  ​$model_path​ ​$image_path
```

**Comment by lvhan028** [comment]
Hi, @DDGRCF. Please show us the exact way to reproduce your issue. I am afraid `$model_path` and `$image_path` cannot provide any information to us. 

Could you let us know the command you use to convert a model? And it will be a great favor if you can post the result of `python tools/check_env.py`.

**Comment by lvhan028** [comment]
Closing it since no activity for a long time. Feel free to reopen it if it is still an issue

**Comment by SongYii** [comment]
I have encountered the same problem, is there a solution?

### Error
 ./pose_detection cpu ../../mmdeploy_model/hrnet/ ../../demo/resources/human-pose.jpg
`[2023-01-10 16:26:24.589] [mmdeploy] [info] [model.cpp:35] [DirectoryModel] Load model: "../../mmdeploy_model/hrnet/"
[2023-01-10 16:26:25.105] [mmdeploy] [info] [inference.cpp:44] {
  "context": {
    "device": "<any>",
    "model": "<any>",
    "stream": "<any>"
  },
  "pipeline": {
    "input": [
      "img"
    ],
    "output": [
      "post_output"
    ],
    "tasks": [
      {
        "input": [
          "img"
        ],
        "module": "Transform",
        "name": "Preprocess",
        "output": [
          "prep_output"
        ],
        "transforms": [
          {
            "type": "LoadImageFromFile"
          },
          {
            "image_size": [
              192,
              256
            ],
            "padding": 1.25,
            "type": "TopDownGetBboxCenterScale"
          },
          {
            "image_size": [
              192,
              256
            ],
            "type": "TopDownAffine"
          },
          {
            "mean": [
              123.675,
              116.28,
              103.53
            ],
            "std": [
              58.395,
              57.120000000000005,
              57.375
            ],
            "to_rgb": true,
            "type": "Normalize"
          },
          {
            "keys": [
              "img"
            ],
            "type": "ImageToTensor"
          },
          {
            "keys": [
              "img"
            ],
            "meta_keys": [
              "flip_direction",
              "rotation",
              "img_shape",
              "filename",
              "ori_filename",
              "ori_shape",
              "flip",
              "img_norm_cfg",
              "center",
              "scale_factor",
              "valid_ratio",
              "image_file",
              "pad_shape",
              "scale",
              "bbox_score",
              "flip_pairs"
            ],
            "type": "Collect"
          }
        ],
        "type": "Task"
      },
      {
        "input": [
          "prep_output"
        ],
        "input_map": {
          "img": "input"
        },
        "is_batched": false,
        "module": "Net",
        "name": "topdown",
        "output": [
          "infer_output"
        ],
        "output_map": {},
        "type": "Task"
      },
      {
        "component": "TopdownHeatmapSimpleHeadDecode",
        "input": [
          "prep_output",
          "infer_output"
        ],
        "module": "mmpose",
        "name": "postprocess",
        "output": [
          "post_output"
        ],
        "params": {
          "flip_test": true,
          "modulate_kernel": 11,
          "post_process": "default",
          "shift_heatmap": true
        },
        "type": "Task"
      }
    ]
  }
}
[2023-01-10 16:26:49.166] [mmdeploy] [info] [inference.cpp:56] ["img"] <- ["image"]
[2023-01-10 16:26:49.167] [mmdeploy] [info] [inference.cpp:67] ["post_output"] -> ["dets"]
/opt/rh/devtoolset-9/root/usr/include/c++/9/bits/stl_vector.h:1042: std::vector<_Tp, _Alloc>::reference std::vector<_Tp, _Alloc>::operator[](std::vector<_Tp, _Alloc>::size_type) [with _Tp = int; _Alloc = std::allocator<int>; std::vector<_Tp, _Alloc>::reference = int&; std::vector<_Tp, _Alloc>::size_type = long unsigned int]: Assertion '__builtin_expect(__n < this->size(), true)' failed.
Aborted (core dumped)`

### Envs
`/home/ps/anaconda3/envs/yisong_dis/lib/python3.9/site-packages/mmcv/__init__.py:20: UserWarning: On January 1, 2023, MMCV will release v2.0.0, in which it will remove components related to the training process and add a data transformation module. In addition, it will rename the package names mmcv to mmcv-lite and mmcv-full to mmcv. See https://github.com/open-mmlab/mmcv/blob/master/docs/en/compatibility.md for more details.
  warnings.warn(
2023-01-10 16:35:40,065 - mmdeploy - INFO -

2023-01-10 16:35:40,065 - mmdeploy - INFO - **********Environmental information**********
2023-01-10 16:35:41,427 - mmdeploy - INFO - sys.platform: linux
2023-01-10 16:35:41,428 - mmdeploy - INFO - Python: 3.9.15 (main, Nov 24 2022, 14:31:59) [GCC 11.2.0]
2023-01-10 16:35:41,428 - mmdeploy - INFO - CUDA available: True
2023-01-10 16:35:41,428 - mmdeploy - INFO - GPU 0: NVIDIA GeForce RTX 3090 Ti
2023-01-10 16:35:41,428 - mmdeploy - INFO - CUDA_HOME: /home/ps/anaconda3/envs/yisong_dis
2023-01-10 16:35:41,428 - mmdeploy - INFO - NVCC: Cuda compilation tools, release 11.7, V11.7.99
2023-01-10 16:35:41,428 - mmdeploy - INFO - GCC: gcc (Ubuntu 11.3.0-1ubuntu1~22.04) 11.3.0
2023-01-10 16:35:41,428 - mmdeploy - INFO - PyTorch: 1.13.1
2023-01-10 16:35:41,428 - mmdeploy - INFO - PyTorch compiling details: PyTorch built with:
  - GCC 9.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.6.0 (Git Hash 52b5f107dd9cf10910aaa19cb47f3abf9b349815)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 11.7
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_80,code=sm_80;-gencode;arch=compute_86,code=sm_86;-gencode;arch=compute_37,code=compute_37
  - CuDNN 8.5
  - Magma 2.6.1
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=11.7, CUDNN_VERSION=8.5.0, CXX_COMPILER=/opt/rh/devtoolset-9/root/usr/bin/c++, CXX_FLAGS= -fabi-version=11 -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Werror=non-virtual-dtor -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wunused-local-typedefs -Wno-unused-parameter -Wno-unused-function -Wno-unused-result -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Werror=cast-function-type -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.13.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, USE_ROCM=OFF,

2023-01-10 16:35:41,428 - mmdeploy - INFO - TorchVision: 0.14.1
2023-01-10 16:35:41,428 - mmdeploy - INFO - OpenCV: 4.5.4
2023-01-10 16:35:41,428 - mmdeploy - INFO - MMCV: 1.7.1
2023-01-10 16:35:41,428 - mmdeploy - INFO - MMCV Compiler: GCC 9.3
2023-01-10 16:35:41,428 - mmdeploy - INFO - MMCV CUDA Compiler: 11.7
2023-01-10 16:35:41,428 - mmdeploy - INFO - MMDeploy: 0.12.0+e4ad0d4
2023-01-10 16:35:41,428 - mmdeploy - INFO -

2023-01-10 16:35:41,428 - mmdeploy - INFO - **********Backend information**********
2023-01-10 16:35:42,044 - mmdeploy - INFO - tensorrt:	None
2023-01-10 16:35:42,755 - mmdeploy - INFO - ONNXRuntime:	1.8.1
2023-01-10 16:35:42,755 - mmdeploy - INFO - ONNXRuntime-gpu:	1.13.1
2023-01-10 16:35:42,756 - mmdeploy - INFO - ONNXRuntime custom ops:	Available
2023-01-10 16:35:42,892 - mmdeploy - INFO - pplnn:	None
2023-01-10 16:35:42,976 - mmdeploy - INFO - ncnn:	None
2023-01-10 16:35:42,980 - mmdeploy - INFO - snpe:	None
2023-01-10 16:35:43,495 - mmdeploy - INFO - openvino:	2022.3.0
2023-01-10 16:35:43,512 - mmdeploy - INFO - torchscript:	1.13.1
2023-01-10 16:35:43,513 - mmdeploy - INFO - torchscript custom ops:	NotAvailable
2023-01-10 16:35:43,657 - mmdeploy - INFO - rknn-toolkit:	None
2023-01-10 16:35:43,657 - mmdeploy - INFO - rknn2-toolkit:	None
2023-01-10 16:35:43,683 - mmdeploy - INFO - ascend:	None
2023-01-10 16:35:43,688 - mmdeploy - INFO - coreml:	None
2023-01-10 16:35:43,850 - mmdeploy - INFO - tvm:	None
2023-01-10 16:35:43,850 - mmdeploy - INFO -

2023-01-10 16:35:43,850 - mmdeploy - INFO - **********Codebase information**********
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmdet:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmseg:	0.29.1
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmcls:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmocr:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmedit:	0.16.0
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmdet3d:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmpose:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmrotate:	None
2023-01-10 16:35:43,875 - mmdeploy - INFO - mmaction:	None`
